pkgs/by-name/ca/cavalcade/package.nix

```diff
@@ -0,0 +1,72 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  gtk3,
+  cava,
+  gst_all_1,
+  gobject-introspection,
+  wrapGAppsHook3,
+  makeDesktopItem,
+  fetchurl,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "cavalcade";
+  version = "0.8";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "worron";
+    repo = "cavalcade";
+    rev = "refs/tags/${version}";
+    hash = "sha256-VyWOPNidN0+pfuxsgPWq6lI5gXQsiRpmYjQYjZW6i9w=";
+  };
+
+  postPatch = ''
+    substituteInPlace cavalcade/cava.py \
+      --replace-fail '"cava"' '"${cava}/bin/cava"'
+  '';
+
+  build-system = [ python3Packages.setuptools ];
+
+  dependencies = with python3Packages; [
+    pygobject3
+    gst-python
+    pillow
+  ];
+
+  nativeBuildInputs = [
+    wrapGAppsHook3
+    gobject-introspection
+    gst_all_1.gstreamer
+  ];
+
+  buildInputs = [ gtk3 ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Cavalcade";
+      type = "Application";
+      exec = "cavalcade";
+      icon = fetchurl {
+        url = "https://raw.githubusercontent.com/worron/cavalcade/devel/desktop/cavalcade.svg";
```

Comment:
I assume `devel` is not a stable target and may break the hash. Please fix the url with a commit id instead